### ISSUE TITLE
Why the PPO tf agent implemented by google only accepts the observation space to be a "gym box space"? 

### ISSUE THREAD

**Issue by navidyou** [tensorflow/agents#623]
I have tried the ppo agent with different observation spaces. It turns out it "only" works with a gym box space. 

---Choosing the observation space as array_spec.BoundedArraySpec( shape=(1,), dtype=np.int32, minimum=0, maximum=32, name='observation') gives an error saying "Cast string to float is not supported".

---Choosing the observation space as a discrete gym space space.Tuple again gives an error saying "Cast string to float is not supported"

To reproduce the second error simply open the file of ppo example V2 train_eval_clip_agent.py https://github.com/tensorflow/agents/blob/master/tf_agents/agents/ppo/examples/v2/train_eval_clip_agent.py and change line 87 from env_name='HalfCheetah-v2' to env_name='Blackjack-v0' The blackjack env implemented in gym has the following observation space spaces.Tuple((
            spaces.Discrete(32),
            spaces.Discrete(11),
            spaces.Discrete(2)))

How can the 'Blackjack-v0' implemented in open AI gym be altered to be usable by TensorFlow PPO agent, or how can the PPO agent be altered so it can be used to train with the gym Blackjack-v0?

Thank you so much for your answer!
Best,
Navid

**Comment by navidyou** [comment]
Sorry for my mistake the observation space for ppo tf_agent can be a tuple of spaces.Discrete as well. 

In the PPO example V2 in order to use the gym 'Blackjack-v0' environment the preprocessing_layers and the preprocessing_combiner must be set. 
By looking at the observation space spec for the gym 'Blackjack-v0' the preprocessing should be set as follows:

`      actor_net = actor_distribution_network.ActorDistributionNetwork(
          tf_env.observation_spec(),
          tf_env.action_spec(),
          (tf.keras.layers.experimental.preprocessing.CategoryEncoding(num_tokens=32, output_mode="binary"),
            tf.keras.layers.experimental.preprocessing.CategoryEncoding(num_tokens=11, output_mode="binary"), 
            tf.keras.layers.experimental.preprocessing.CategoryEncoding(num_tokens=2, output_mode="binary")),
          tf.keras.layers.Concatenate(axis=-1),
          #fc_layer_params=actor_fc_layers,
          activation_fn=tf.keras.activations.tanh)
      value_net = value_network.ValueNetwork(
          tf_env.observation_spec(),
          (tf.keras.layers.experimental.preprocessing.CategoryEncoding(num_tokens=32, output_mode="binary"),
            tf.keras.layers.experimental.preprocessing.CategoryEncoding(num_tokens=11, output_mode="binary"), 
            tf.keras.layers.experimental.preprocessing.CategoryEncoding(num_tokens=2, output_mode="binary")),
          tf.keras.layers.Concatenate(axis=-1),
          #fc_layer_params=value_fc_layers,
          activation_fn=tf.keras.activations.tanh)`
